File: Sample/Engine/Modified/RenderStates.py
# -*- encoding: utf-8 -*-

from pysf import RenderStates, BlendMode


class ModifiedRenderStates(RenderStates):
    r"""
    \brief Modified Render States.
    """

    @staticmethod
    def Default() -> RenderStates:
        r"""
        \brief Default.
        """

        return RenderStates(
            BlendMode(
                BlendMode.Factor.SrcAlpha,
                BlendMode.Factor.OneMinusSrcAlpha,
                BlendMode.Equation.Add,
                BlendMode.Factor.One,
                BlendMode.Factor.OneMinusSrcAlpha,
                BlendMode.Equation.Add,
            )
        )

File: Game/Engine/Source/Core/src/Modified/RenderStates.cpp
#include <Modified/RenderStates.hpp>

#include <SFML/Graphics/BlendMode.hpp>

namespace {

sf::BlendMode canvasBlendMode() {
    return sf::BlendMode(
        sf::BlendMode::Factor::SrcAlpha,
        sf::BlendMode::Factor::OneMinusSrcAlpha, sf::BlendMode::Equation::Add,
        sf::BlendMode::Factor::One, sf::BlendMode::Factor::OneMinusSrcAlpha,
        sf::BlendMode::Equation::Add);
}

}  // namespace

ModifiedRenderStates::ModifiedRenderStates(const sf::RenderStates& states)
    : sf::RenderStates(states) {}

ModifiedRenderStates ModifiedRenderStates::Default() {
    return ModifiedRenderStates(sf::RenderStates(canvasBlendMode()));
}
